src/main/java/com/griddynamics/collectionsTask/CollectionsTask.java

```diff
@@ -0,0 +1,190 @@
+        } catch (FileFormatException | IOException e) {
+            e.printStackTrace();
+        }
+        merge(cars, trucks, vehicles);
+        sortListOfVehicles(vehicles);
+        System.out.println(vehicles);
+        fillMap(vehicles, producers);
+        System.out.println(producers);
+    }
+
+    public static void sortListOfVehicles(List<Vehicle> vehicles) {
+        vehicles.sort(Vehicle::compareTo);
+    }
+
+    public static void fillMap(List<Vehicle> vehicles, Map<String, List<Vehicle>> producers) {
+        for (Vehicle vehicle : vehicles) {
+            if (!producers.containsKey(vehicle.producer)) {
+                producers.put(vehicle.producer, new ArrayList<>());
+            }
+            producers.get(vehicle.producer).add(vehicle);
+        }
+    }
+
+    public static void merge(List<Car> cars, List<Truck> trucks, List<Vehicle> vehicles) {
+        vehicles.addAll(cars);
+        vehicles.addAll(trucks);
+    }
+
+    public static void read(List<Car> cars, List<Truck> trucks, String inputFile) throws FileFormatException, IOException {
+        Path inputFilePath = Paths.get(inputFile);
+        List<String> lines = Files.readAllLines(inputFilePath);
+        for (int i = 0; i < lines.size(); i++) {
+            String line = lines.get(i);
+            String cur = line.replaceAll("\\s+", "");
+            cur = cur.replace(':', ',');
+            String[] parameters = cur.split(",");
+            if (parameters.length == 5) {
+                if (parameters[0].equalsIgnoreCase("CAR")) {
+                    cars.add(new Car(parameters[1], parameters[2],
+                            Integer.parseInt(parameters[3]), parameters[4]));
+                } else {
+                    trucks.add(new Truck(parameters[1], parameters[2],
+                            Integer.parseInt(parameters[3]), Long.parseLong(parameters[4])));
+                }
+            } else {
+                throw new FileFormatException("Your values in file are invalid. " +
+                        "Check them and try again.", i + 1);
+            }
+        }
+        if ((cars != null && cars.isEmpty()) && (trucks != null && trucks.isEmpty())) {
+            System.out.println("Empty file provided.");
+        }
+    }
+
+    public static class FileFormatException extends Exception {
+        private final int lineNumber;
+
+        public FileFormatException(String message, int lineNumber) {
+            super(message);
+            this.lineNumber = lineNumber;
+        }
+
+        public int getLineNumber() {
+            return lineNumber;
+        }
+    }
+
+    public static class Vehicle {
+        protected String model;
+        protected String producer;
+        protected int age;
+
+        public Vehicle(String model, String producer, int age) {
+            this.model = model;
+            this.producer = producer;
+            this.age = age;
+        }
+
+        public String getModel() {
```

Comment:
@kolegova-mary do not need to generate all getters/setters if you do not use them, you could create only those that are required